qiskit_ibm_provider/ibm_backend.py

```diff
@@ -520,8 +520,11 @@ def _get_run_config(self, program_id: str, **kwargs: Any) -> Dict:
             fields = asdict(QASM2Options()).keys()
             run_config_dict = QASM2Options().to_transport_dict()
 
+        backend_options = self._options.__dict__
         for key, val in kwargs.items():
-            if val is not None:
+            if val is None and backend_options.get(key) is not None and key in fields:
+                run_config_dict[key] = backend_options[key]
+            elif val is not None:
                 run_config_dict[key] = val
```

Comment:
My understanding is that we should only add the backend options value to the `run_config_dict` if it exists and if the kwargs value is None (to avoid overwriting).

The key itself should be in fields to avoid "unexpected keyword argument" in `qasm3-runner` jobs